frontend/src/components/common/Button.vue

```diff
@@ -1,16 +1,24 @@
-<script setup>
+<script setup lang="ts">
 import Button from 'primevue/button';
+import type { PropType } from 'vue';
+import { IconPosition } from '../../enum/IconEnum';
 const props = defineProps({
     label: {
         type: String,
         default: 'Click',
     },
+    iconPos: {
+        type: String as PropType<IconPosition>,
+        default: IconPosition.right,
+    },
```

Comment:
I'm not so sure why we need this iconPos property?

Reply:
Hello, Catherine! The sign out button has the icon on a before the text, to use this component I did this so we could just pass the side we want the icon

Reply:
Oh, I got why we need it, for the icon